tests/sles4sap/patterns.pm

```diff
@@ -29,24 +29,30 @@ sub run {
     quit_packagekit;
 
     # Is HA pattern needed?
-    push @sappatterns, 'ha_sles' if get_var('HA_CLUSTER');
+    if (get_var('HA_CLUSTER')) {
+        is_sle('16+') ? push(@sappatterns, 'sles_sap_HAAPP', 'sles_sap_HADB') : push(@sappatterns, 'ha_sles');
```

Comment:
16 also has `sles-ha`. Did you check whether this is pulled by `sles_sap_HAAPP` or by `sles_sap_HADB` or if it should be explicitly installed?

Reply:
Sorry, I misunderstood this pattern.  According to [pattern discussion](https://confluence.suse.com/display/SAP/SLES4SAP+16+pattern+discussion), maybe when enable HA_CLUSTER, we need to install `sles_sap_HAAPP` and `sles_sap_HADB`. So I add them here.

Reply:
Yes, that's OK. But I think we also need to install `sles-ha`, unless it's included in the other 2 HA patterns.